src/context/AppProvider/AppContext.tsx

```diff
@@ -161,11 +166,20 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
 
   // once portfolio is done loading, fetch all transaction history
   useEffect(() => {
-    if (portfolioLoadingStatus === 'loading') return
+    ;(async () => {
+      if (portfolioLoadingStatus === 'loading') return
 
-    const { getAllTxHistory } = txHistoryApi.endpoints
+      const { getAllTxHistory } = txHistoryApi.endpoints
 
-    requestedAccountIds.forEach(accountId => dispatch(getAllTxHistory.initiate(accountId)))
+      try {
+        await Promise.all(
+          requestedAccountIds.map(accountId => dispatch(getAllTxHistory.initiate(accountId))),
+        )
+      } finally {
+        const txsById = store.getState().txHistory.txs.byId
```

Comment:
fetch all transactions from the store to detect and upsert any nft assets.

Reply:
Maybe add a comment why we don't `useAppSelector()` on this guy?

Reply:
I assume this is just a fire-and-forget bg trigger fetch that doesn't block UI, correct? Also, looks like this might trigger multiple times even once it's been loaded

Reply:
adding comment